src/python/foglamp/configuration_manager.py

```diff
@@ -271,16 +283,31 @@
     item_name -- name of item within the category whose "value" entry needs to be changed (required)
     new_value_entry -- new value entry to replace old value entry
 
+    Side Effects:
+    An update to storage will not be issued if a new_value_entry is the same as the new_value_entry from storage.
+    Registered callbacks will be invoked only if an update is issued.
+
     Return Values:
     None
     """
     try:
-        return await _update_value_val(category_name, item_name, new_value_entry)
+        # get storage_value_entry and compare against new_value_value, update if different
+        storage_value_entry = await _read_value_val(category_name, item_name)
+        if storage_value_entry == new_value_entry:
+            # raise ValueError('new_value_entry must be different than value_entry in storage')
+            return
```

Comment:
When the value in storage is the same as the new value, I do not make an update to the DB. In these cases, should I raise an exception, or return with no message?

Reply:
I think silent return is fine